front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -30,30 +37,147 @@ import {
   isManaged,
   isWebsite,
 } from "@app/lib/data_sources";
+import { useVaults } from "@app/lib/swr/vaults";
+import { classNames } from "@app/lib/utils";
+import { getVaultIcon, getVaultName } from "@app/lib/vaults";
 
 const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
 const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+const ONLY_ONE_VAULT_PER_SELECTION = true;
 
 interface DataSourceViewsSelectorProps {
   owner: LightWorkspaceType;
   dataSourceViews: DataSourceViewType[];
+  allowedVaults?: VaultType[];
   selectionConfigurations: DataSourceViewSelectionConfigurations;
   setSelectionConfigurations: Dispatch<
     SetStateAction<DataSourceViewSelectionConfigurations>
   >;
   viewType: ContentNodesViewType;
 }
 
+function VaultSelector({
+  owner,
+  dataSourceViews,
+  allowedVaults,
+  selectionConfigurations,
+  setSelectionConfigurations,
+  viewType,
+}: DataSourceViewsSelectorProps) {
+  const dataSourceViewsByVaultId = useMemo(
+    () => _.groupBy(dataSourceViews, (dsv) => dsv.vaultId),
+    [dataSourceViews]
+  );
+
+  const { vaults, isVaultsError, isVaultsLoading } = useVaults({
+    workspaceId: owner.sId,
+  });
+
+  const [selectedVault, setSelectedVault] = useState<string>(
+    Object.keys(selectionConfigurations).length > 0
+      ? selectionConfigurations[Object.keys(selectionConfigurations)[0]]
+          .dataSourceView.vaultId
+      : ""
+  );
+
+  if (isVaultsLoading || isVaultsError || !vaults) {
+    return <Spinner />;
+  }
+
+  if (Object.keys(dataSourceViewsByVaultId).length === 1) {
+    return (
+      <DataSourceViewsSelector
+        owner={owner}
+        dataSourceViews={dataSourceViews}
+        selectionConfigurations={selectionConfigurations}
+        setSelectionConfigurations={setSelectionConfigurations}
+        viewType={viewType}
+      />
+    );
+  } else {
+    return (
+      <div>
+        <Page.Separator />
+        {Object.keys(dataSourceViewsByVaultId).map((vaultId) => {
+          const vault = vaults.find((v) => v.sId === vaultId);
+          if (!vault) {
+            // Should never happen
+            return null;
+          }
+          const disabled = !(allowedVaults
+            ? allowedVaults.find((v) => v.sId === vaultId)
+            : false);
+          return (
+            <div key={vaultId}>
+              <RadioButton
+                name={`Vault ${vaultId}`}
+                choices={[
+                  {
+                    label: (
+                      <>
+                        <Icon
+                          visual={getVaultIcon(vault)}
+                          size="md"
+                          className="s-flex-shrink-0 ml-3 inline-block align-middle text-brand"
```

Comment:
Tailwind classes prefixed with `s-` should only be used in Sparkle 🙃 
```suggestion
                          className="flex-shrink-0 ml-3 inline-block align-middle text-brand"
```